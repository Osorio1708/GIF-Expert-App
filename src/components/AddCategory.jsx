import { useState } from "react";

export const AddCategory = ({
  //onAddCategory
  onNewCategory,
}) => {
  const [inputValue, setInputValue] = useState("");

  //   const onInputChange = (text) => {
  //     setInputValue(text);
  //   };

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    // console.log(event);
    // Previene que se recargue la p√°gina

    event.preventDefault();
    const value = inputValue.trim();
    if (value.length <= 1) return;

    //onAddCategory((categories) => [...categories, inputValue]);

    onNewCategory(value);
    setInputValue("");
  };

  return (
    // <form onSubmit={(event) => onSubmit(event)}>

    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
