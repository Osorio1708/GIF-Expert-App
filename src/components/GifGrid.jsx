import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);
  // const [images, setImages] = useState([]);

  // const getImages = async () => {
  //   const newImages = await getGifs(category);
  //   setImages(newImages);
  // };

  // useEffect(() => {
  //   getImages();
  // }, []);

  return (
    <>
      <h3>{category}</h3>
      {/* {isLoading ? <h2>Cargando...</h2> : null} */}
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {images.map((img) => (
          <GifItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
