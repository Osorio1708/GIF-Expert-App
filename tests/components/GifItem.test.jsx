import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test of <GifItem />", () => {
  const title = "L";
  const url = "https://death-note/L.png";

  test("should match with snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show image with respective URL and ALT", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show component title", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
