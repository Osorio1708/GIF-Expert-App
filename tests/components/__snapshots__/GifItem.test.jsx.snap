// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="L"
      src="https://death-note/L.png"
    />
    <p>
      L
    </p>
  </div>
</div>
`;
